src/track/beatutils.h

```diff
@@ -68,6 +74,13 @@ class BeatUtils {
         const QVector<double> rawbeats, const int sampleRate,
         const int totalSamples, const double globalBpm);
 
+    static double samplesToFrames(const double samples) {
+        return floor(samples / mixxx::kEngineChannelCount);
+    }
+    static double framesToSamples(const int frames) {
```

Comment:
The implicit type conversion from int to double is confusing. Please use either integer or floating point but not both. Use separate or specialized functions for integer rounding operations.